src/utils/use-interval.ts

```diff
@@ -17,15 +18,32 @@ export function useInterval(
   // Set up the interval.
   useEffect(() => {
     if (disabled) return;
+    const actualDelay = Math.max(delay ?? 0, 100);
+
+    let id: NodeJS.Timeout | number = 0;
+
     function tick() {
-      savedCallback.current();
-    }
-    if (triggerImmediately) {
-      tick();
+      return setTimeout(async function innerCallback() {
+        await savedCallback.current();
+
+        if (!disabled) {
+          tick();
+        }
+      }, actualDelay);
     }
-    if (delay !== null) {
-      let id = setInterval(tick, delay);
-      return () => clearInterval(id);
+
+    async function invoke() {
+      if (triggerImmediately) {
+        await savedCallback.current();
+      }
+      id = tick();
     }
+
+    invoke();
+    return () => {
+      if (id) {
+        clearTimeout(id as number);
```

Comment:
Was a little unsure about if this would clear recursively created `setTimout` invocations, so I tried a quick test in the console, and it seems like it doesn't. Maybe create an `timeoutIdRef` or something?

![Screenshot 2023-09-01 at 12 10 20](https://github.com/AtB-AS/mittatb-app/assets/675421/c6c0f28b-77dc-4979-a3c6-4ba1d9588324)

Reply:
Hmm. Yeah, thats a good point 🤔 might refer to different value there

Reply:
Or just set `id = tick()` in the recursive call as well🤦:sweat_smile:

Reply:
Haha, yes. That would work better 😅